pkg/osbuild/source_test.go

```diff
@@ -217,3 +241,100 @@ func TestGenSourcesWithSkopeoIndex(t *testing.T) {
   }
 }`)
 }
+
+// TODO: move into a common "rpmtest" package
+var opensslPkg = rpmmd.PackageSpec{
+	Name:           "openssl-libs",
+	RemoteLocation: "https://example.com/repo/Packages/openssl-libs-3.0.1-5.el9.x86_64.rpm",
+	Checksum:       "sha256:fcf2515ec9115551c99d552da721803ecbca23b7ae5a974309975000e8bef666",
+	Path:           "Packages/openssl-libs-3.0.1-5.el9.x86_64.rpm",
+	RepoID:         "repo_id_metalink",
+}
+
+var fakeRepos = []rpmmd.RepoConfig{
+	{
+		Id:       "repo_id_metalink",
+		Metalink: "http://example.com/metalink",
+	},
+}
+
+func TestGenSourcesRpmDefaultRpmDownloaderIsCurl(t *testing.T) {
+	inputs := SourceInputs{
+		Depsolved: dnfjson.DepsolveResult{
+			Packages: []rpmmd.PackageSpec{opensslPkg},
+			Repos:    fakeRepos,
+		},
+	}
+	var defaultRpmDownloader RpmDownloader
+	sources, err := GenSources(inputs, defaultRpmDownloader)
+	assert.NoError(t, err)
+
+	assert.NotNil(t, sources["org.osbuild.curl"])
+	assert.Nil(t, sources["org.osbuild.librepo"])
```

Comment:
Nitpick: it would make sense to use the newly defined source name constants in tests as well, but I won't block on it.